Geometry/GEMGeometryBuilder/src/ME0GeometryBuilder.cc

```diff
@@ -0,0 +1,544 @@
+
+#ifdef EDM_ML_DEBUG
+  LogTrace("ME0GeometryBuilder") << " name of logical part = " << fv.logicalPart().name().name() << std::endl;
+  LogTrace("ME0GeometryBuilder") << " dpar is vector with size = " << dpar.size() << std::endl;
+  for (unsigned int i = 0; i < dpar.size(); ++i) {
+    LogTrace("ME0GeometryBuilder") << " dpar [" << i << "] = " << dpar[i] / 10 << " cm " << std::endl;
+  }
+  LogTrace("ME0GeometryBuilder") << "size  b: " << b << "cm, B: " << B << "cm,  L: " << L << "cm, T: " << T << "cm "
+                                 << std::endl;
+#endif
+
+  bool isOdd = false;  // detId.chamber()%2;
+  ME0BoundPlane surf(boundPlane(fv, new TrapezoidalPlaneBounds(b, B, L, T), isOdd));
+  ME0Chamber* chamber = new ME0Chamber(detId.chamberId(), surf);
+  return chamber;
+}
+
+ME0Layer* ME0GeometryBuilder::buildLayer(DDFilteredView& fv, ME0DetId detId) const {
+  LogTrace("ME0GeometryBuilder") << "buildLayer " << fv.logicalPart().name().name() << " " << detId << std::endl;
+
+  DDBooleanSolid solid = (DDBooleanSolid)(fv.logicalPart().solid());
+  // std::vector<double> dpar = solid.solidA().parameters();
+  std::vector<double> dpar = solid.parameters();
+  double L = dpar[0] / cm;  // length is along local Y
+  double t = dpar[3] / cm;  // thickness is long local Z
+  double b = dpar[4] / cm;  // bottom width is along local X
+  double B = dpar[8] / cm;  // top width is along local X
+  // dpar = solid.solidB().parameters();
+  // dz += dpar[3]/cm;     // layer thickness --- to be checked !!! layer thickness should be same as eta part thickness
+  // hardcoded :: double b = 21.9859, B = 52.7261, L = 87.1678, t = 0.4;
+
+#ifdef EDM_ML_DEBUG
+  LogTrace("ME0GeometryBuilder") << " name of logical part = " << fv.logicalPart().name().name() << std::endl;
+  LogTrace("ME0GeometryBuilder") << " dpar is vector with size = " << dpar.size() << std::endl;
+  for (unsigned int i = 0; i < dpar.size(); ++i) {
+    LogTrace("ME0GeometryBuilder") << " dpar [" << i << "] = " << dpar[i] / 10 << " cm " << std::endl;
+  }
+  LogTrace("ME0GeometryBuilder") << "size  b: " << b << "cm, B: " << B << "cm,  L: " << L << "cm, t: " << t << "cm "
+                                 << std::endl;
+#endif
+
+  bool isOdd = false;  // detId.chamber()%2;
+  ME0BoundPlane surf(boundPlane(fv, new TrapezoidalPlaneBounds(b, B, L, t), isOdd));
+  ME0Layer* layer = new ME0Layer(detId.layerId(), surf);
+  return layer;
+}
+
+ME0EtaPartition* ME0GeometryBuilder::buildEtaPartition(DDFilteredView& fv, ME0DetId detId) const {
+  LogTrace("ME0GeometryBuilder") << "buildEtaPartition " << fv.logicalPart().name().name() << " " << detId << std::endl;
+
+  // EtaPartition specific parameter (nstrips and npads)
+  DDValue numbOfStrips("nStrips");
+  DDValue numbOfPads("nPads");
+  std::vector<const DDsvalues_type*> specs(fv.specifics());
+  std::vector<const DDsvalues_type*>::iterator is = specs.begin();
+  double nStrips = 0., nPads = 0.;
+  for (; is != specs.end(); is++) {
```

Comment:
range-based loop?

Reply:
@kpedro88 : this is the loop for retrieve information about nStrips and nPad (DDD version)

Reply:
https://en.cppreference.com/w/cpp/language/range-for

Reply:
@kpedro88 : done